engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java

```diff
@@ -182,17 +183,30 @@ Type getType() {
   }
 
   @ExportMessage
-  boolean hasMetaObject() {
+  boolean hasMetaObject(@CachedLibrary("this") InteropLibrary lib) {
+    var b = EnsoContext.get(lib).getBuiltins();
+    if (this == b.nothing()) {
+      return false;
+    }
```

Comment:
Since that change, I'm slightly worried if `Nothing` will correctly work with our types? Hopefully yes, but to make us confident I'd recommend to add the following tests somewhere around `Meta_Spec.enso`:

```
Nothing.is_a Nothing . should_be_true
Meta.type_of Nothing . should_equal Nothing
```

These are currently passing, let's make sure we don't regress over them.

Reply:
Yes, I also find this change a bit suspicions. Do you recall what triggered that?

Reply:
I think it failed in assertions in `InteropLibrary`, specifically in `InteropLibrary.noOtherType` (https://github.com/oracle/graal/blob/master/truffle/src/com.oracle.truffle.api.interop/src/com/oracle/truffle/api/interop/InteropLibrary.java#L2965 ). Seems that Truffle expects that if something is a null (from the interop perspective), then it cannot also be a meta object.

Reply:
null cannot be a meta object at the same time, but it could _have_ a meta object

We could do
```
Meta.type_of Nothing == Nothing_Type
```

Not necessarily what we want, but that's one of possible solutions

Reply:
ah, right. `Nothing` vs `Nothing.type` that I've stumbled upon myself in https://github.com/enso-org/enso/pull/4077

Reply:
> ah, right. `Nothing` vs (lack of) `Nothing.type` that I've stumbled upon myself in #4077

Yep, that's what I was mentioning in the comment to that PR of yours.

Reply:
Good idea to add the tests. Done in 1e57210d6c185b - they pass OK. The change of `getMetaObject` only affects interop - not internal Enso behavior as Enso engine is using `TypesLibrary` (if available for given value) and there is no change in `TypesLibrary` for `Nothing`.

Reply:
Great to hear that they work! Thanks for verifying it.